File: square-commons/src/main/java/com/raxdenstudios/square/interceptor/commons/fragmentstatepager/FragmentStatePagerInterceptorCallback.java
package com.raxdenstudios.square.interceptor.commons.fragmentstatepager;

import android.os.Bundle;

import com.raxdenstudios.square.interceptor.InterceptorCallback;

import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

/**
 * Created by Ángel Gómez on 20/12/2016.
 */
public interface FragmentStatePagerInterceptorCallback<TFragment extends Fragment> extends InterceptorCallback {

    ViewPager onCreateViewPager(Bundle savedInstanceState);

    void onViewPagerCreated(ViewPager viewPager);

    TFragment onCreateFragment(int position);

    void onFragmentLoaded(TFragment fragment, int position);

    void onFragmentSelected(TFragment fragment, int position);

    int getViewPagerElements();

}

File: square-commons/src/main/java/com/raxdenstudios/square/interceptor/commons/fragmentstatepager/FragmentStatePagerInterceptorCallback.kt
package com.raxdenstudios.square.interceptor.commons.fragmentstatepager

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.view.ViewPager

import com.raxdenstudios.square.interceptor.InterceptorCallback

/**
 * Created by Ángel Gómez on 20/12/2016.
 */
interface FragmentStatePagerInterceptorCallback<TFragment : Fragment> : InterceptorCallback {

    val viewPagerElements: Int

    fun onCreateViewPager(savedInstanceState: Bundle?): ViewPager

    fun onViewPagerCreated(viewPager: ViewPager)

    fun onCreateFragment(position: Int): TFragment?

    fun onFragmentLoaded(fragment: TFragment?, position: Int)

    fun onFragmentSelected(fragment: TFragment?, position: Int)

}
